### PR TITLE
Build debian package with FPM

### PR DESCRIPTION
**Pull request by bersace**
This PR propose

- to build .deb from PyPI release instead of master
- to build .deb with public code anyone can contribute
- to test .deb workflow end-to-end with reprepro.

# Testing instructions

1. Read [packaging/README.rst](https://github.com/bersace/temboard-agent/blob/debian-fpm/packaging/README.rst)
1. `rm -rf dist/` at top level.
1. build **all** packages with a single `make build`.
1. Review logs.
1. Follow `README.rst` to run a test APT repository with `make reprepro`.
1. Upload wheezy package with `dput -c dput.cf oldstable ../dist/*_wheezy.changes`
1. Upload jessie package with `dput -c dput.cf stable ../dist/*_jessie.changes`
1. Upload stretch package with `dput -c dput.cf testing ../dist/*_stretch.changes`
1. Test each package following `README.rst`.
   1. Create a guest with the corresponding distribution (e.g. with LXC)
   1. Edit sources.list, add GPG key and update
   1. Install postgresql
   1. Install `temboard-agent`.
   1. run `temboard-agent -c /usr/share/temboard-agent/quickstart/temboard-agent.conf`
1. Ensure old `debian/` and `make_deb.sh` is not referenced elsewhere.

### CODE REVIEW

File: packaging/Makefile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+all:
+
+build: debian-jessie
+
+debian-%:
```
Comment (bersace) [inline packaging/Makefile]: @pgiraud @julmon just `make debian-jessie`, `make debian-stretch`, etc. :-)

File: packaging/README.rst [hunk 2]
```diff
@@ -0,0 +1,38 @@
+=======================================
```
Comment (bersace) [inline packaging/README.rst]: @julmon @pgiraud the starting point is here :-)

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -33,8 +33,11 @@ master as usual.
...
 
-.. code-block
+.. code-block::
 
    git tag 1.1
    git push --tags
    make release
+
+Now build distro packages with ``make -C packaging build``. Ensure you ``NAME``
+and ``EMAIL`` env vars contains maintainer identification.
```
Comment (pgiraud) [inline CONTRIBUTING.rst]: contain

File: packaging/README.rst [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+.. _FPM: https://github.com/jordansissel/fpm
+
+
+For Debian
+==========
+
+The ``mkdeb.sh`` script builds ``.deb`` for the latest version on PyPI. The
+``Makefile`` uses docker to isolate the build environment.
+
+``make deb-$CODENAME`` build the .deb for a specific debian distribution
```
Comment (pgiraud) [inline packaging/README.rst]: builds

File: packaging/README.rst [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+.. notice:: If you get no output just after the run, type ``ENTER`` or
+            ``CTRL+L`` to redraw.
+
+Once you quit the shell, the container is destroyed.
+
+
+APT test repository
+-------------------
+
+``make reprepro`` setup a naïve APT repository. Ensure you have ``.docker``
```
Comment (pgiraud) [inline packaging/README.rst]: naive (no trema in English)

**Comment by julmon** [comment]
Hi,

There is an issue with the debian part (mkdeb.sh), trying it on `jessie` and it seems that some of install paths are not right.

```
# dpkg -c temboard-agent_1.0~a1-0dlb1_all.deb
[...]
-rw-r--r-- 0/0            9677 2017-05-12 14:55 ./destdir/usr/lib/python2.7/dist-packages/temboardagent/sharedmemory.py
-rw-r--r-- 0/0           12135 2017-05-12 14:55 ./destdir/usr/lib/python2.7/dist-packages/temboardagent/inventory.pyc
-rw-r--r-- 0/0            8355 2017-05-12 14:55 ./destdir/usr/lib/python2.7/dist-packages/temboardagent/configuration.pyc
-rw-r--r-- 0/0            2627 2017-05-12 14:55 ./destdir/usr/lib/python2.7/dist-packages/temboardagent/api_wrapper.py
-rw-r--r-- 0/0            1691 2017-05-12 14:55 ./destdir/usr/lib/python2.7/dist-packages/temboardagent/command.pyc
[...]
```

File: packaging/mkdeb.sh [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    --version $debianv \
+    --iteration 0dlb1 \
+    --maintainer "${DEBFULLNAME} <${DEBEMAIL}>" \
+    --license PostgreSQL \
+    --description "Administration & monitoring PostgreSQL agent." \
+    --url http://temboard.io/ \
+    --category database \
+    --depends python2.7 \
+    ${fpm_args[*]} \
+    $DESTDIR=/
```
Comment (julmon) [inline packaging/mkdeb.sh]: You should replace this line by:
```
    -C $DESTDIR
```

This fixes wrong install path like `./destdir/usr/lib/python2.7`

File: packaging/README.rst [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+The ``mkdeb.sh`` script builds ``.deb`` for the latest version on PyPI. The
+``Makefile`` uses docker to isolate the build environment.
+
+``make deb-$CODENAME`` build the .deb for a specific debian distribution
+release: ``stretch``, ``sid``, or any other Debian distribution available at
+https://hub.docker.com/_/buildpack-deps/.
+
+debsign signs ``.changes`` with the GPG signature matching the maintainer field
+``$NAME <$EMAIL>``. You may be prompted for a passphrase to open your private
+key.
```
Comment (pgiraud) [inline packaging/README.rst]: Dependency on debsign (devscripts) needs to be documented.

File: packaging/mkdeb.sh (hunk 6 above)
Comment (bersace) [inline packaging/mkdeb.sh]: Thanks. I hit the bug, i thought i fixed this. Let's squash this again. Thanks for the tip.

**Comment by bersace** [comment]
``` console
root@jessie:/usr/local/src/dalibo/temboard-agent-wip/dist/jessie# dpkg -i temboard-agent_1.0~a1-0dlb1_all.deb
Selecting previously unselected package temboard-agent.
(Reading database ... 24746 files and directories currently installed.)
Preparing to unpack temboard-agent_1.0~a1-0dlb1_all.deb ...
Unpacking temboard-agent (1.0~a1-0dlb1) ...
Setting up temboard-agent (1.0~a1-0dlb1) ...
Created symlink from /etc/systemd/system/multi-user.target.wants/temboard-agent.service to /lib/systemd/system/temboard-agent.service.
Job for temboard-agent.service failed. See 'systemctl status temboard-agent.service' and 'journalctl -xn' for details.
root@jessie:/usr/local/src/dalibo/temboard-agent-wip/dist/jessie# temboard-agent --help
Usage: temboard-agent [options]

temBoard agent.

Options:
  -h, --help            show this help message and exit
  -c CONFIGFILE, --config=CONFIGFILE
                        Configuration file. Default: /etc/temboard-agent
                        /temboard-agent.conf
  -d, --daemon          Run in background. Default: False
  -p PIDFILE, --pid-file=PIDFILE
                        PID file. Default: /run/temboard-agent.pid
root@jessie:/usr/local/src/dalibo/temboard-agent-wip/dist/jessie#
```

:)

**Comment by bersace** [comment]
Let's go and iterate over this.

**Pull request merged**